src/python/pants/backend/go/target_type_rules.py

```diff
@@ -213,6 +228,58 @@ async def inject_go_external_package_dependencies(
     return InjectedDependencies(inferred_dependencies)
 
 
+# -----------------------------------------------------------------------------------------------
+# Generate `_go_external_package` targets
+# -----------------------------------------------------------------------------------------------
+
+
+class GenerateGoExternalPackageTargetsRequest(GenerateTargetsRequest):
+    generate_from = GoModule
+
+
+@rule(desc="Generate targets for each external package in `go.mod`", level=LogLevel.DEBUG)
+async def generate_go_external_package_targets(
+    request: GenerateGoExternalPackageTargetsRequest,
+) -> GeneratedTargets:
+    generator_addr = request.generator.address
+    resolved_module = await Get(ResolvedGoModule, ResolveGoModuleRequest(generator_addr))
+    all_resolved_packages = await MultiGet(
+        Get(
+            ResolveExternalGoModuleToPackagesResult,
+            ResolveExternalGoModuleToPackagesRequest(
```

Comment:
Even if you fix the above cycle, there will be a second one. This one is really weird...the offending rules only use filesystem and Process intrinsics. And fixing involves a weird incantation of commenting out `DigestSubset` and `CreateDigest` `Get`s, but not necessarily one particular one - I had to comment out a few for it to work.